### PR TITLE
Feat: add check is snyk cli installed

### PR DESCRIPTION
**Pull request by aldanchenko**
Added check is Snyk CLI installed. If not installed it will show notification with the link to Snyk CLI installation guide.

74108588-98d3b900-4b84-11ea-8daf-0e653fd04292

### CODE REVIEW

File: src/main/scala/io/snyk/plugin/SnykPluginProjectComponent.scala [hunk 1]
```diff
@@ -20,6 +22,19 @@ class SnykPluginProjectComponent(project: Project) extends ProjectComponent {
   }
 
   override def projectOpened(): Unit = {
+    if (!pluginState.apiClient.isCliInstalled()) {
+      ReportMessages.GROUP.createNotification(
+        "Warning",
+        """Snyk CLI not installed.
```
Comment (yuliabaron) [inline src/main/scala/io/snyk/plugin/SnykPluginProjectComponent.scala]:
```suggestion
        """The Snyk CLI has not been installed.
```

File: src/main/scala/io/snyk/plugin/SnykPluginProjectComponent.scala [hunk 2]
```diff
@@ -20,6 +22,19 @@ class SnykPluginProjectComponent(project: Project) extends ProjectComponent {
...
+          |<a href=\"https://support.snyk.io/hc/en-us/articles/360003812458-Getting-started-with-the-CLI\">
+          |Please, install it
```
Comment (yuliabaron) [inline src/main/scala/io/snyk/plugin/SnykPluginProjectComponent.scala]:
```suggestion
          |See Snyk docs to help you install it.
```

File: src/main/scala/io/snyk/plugin/client/CliClient.scala [hunk 3]
```diff
@@ -21,27 +21,37 @@ import io.circe.{Json, Printer}
...
-sealed trait ApiClient {
+sealed trait CliClient {
   /** Run a scan on the supplied artifact tree */
   def runScan(project: Project, treeRoot: SnykMavenArtifact): Try[Seq[SnykVulnResponse]]
   def userInfo(): Try[SnykUserInfo]
   /** For the "standard" client, returns false if we don't have the necessary credentials */
   def isAvailable: Boolean
+
+  /**
+    * Check is Snyk CLI is installed.
```
Comment (yuliabaron) [inline src/main/scala/io/snyk/plugin/client/CliClient.scala]:
```suggestion
    * Check whether Snyk CLI is installed.
```

File: src/main/scala/io/snyk/plugin/client/CliClient.scala [hunk 4]
```diff
@@ -135,11 +145,30 @@ private final class StandardApiClient(tryConfig: => Try[SnykConfig]) extends Api
...
 
-    val generalCommandLine = new GeneralCommandLine(commands)
-    generalCommandLine.setCharset(Charset.forName("UTF-8"))
-    generalCommandLine.setWorkDirectory(projectPath)
+    new ConsoleCommandRunner().execute(commands, projectPath)
+  }
 
-    ScriptRunnerUtil.getProcessOutput(generalCommandLine, ScriptRunnerUtil.STDOUT_OUTPUT_KEY_FILTER, 120000)
+  def isCliInstalled(consoleCommandRunner: ConsoleCommandRunner = new ConsoleCommandRunner): Boolean = {
+    log.debug("Check is Snyk CLI is installed")
```
Comment (yuliabaron) [inline src/main/scala/io/snyk/plugin/client/CliClient.scala]:
```suggestion
    log.debug("Check whether Snyk CLI is installed")
```

**Comment by aldanchenko** [comment]
@yuliabaron thank you for these important comments! Applied your comments.

**Pull request merged**